R/susie_wrapper.R

```diff
@@ -125,6 +121,114 @@ susie_rss_wrapper <- function(z, R, bhat, shat, n = NULL, var_y = NULL, L = 10,
+          var_y = residual_variance = 1/(phi*(1-phi))
+      }else{
+      # 3. if no infomation related to sample size
+          n = NULL
+      }
+    }     
+    return(rss_input = list(sumstats = sumstats, n = n, var_y = var_y))
+}
+
+
+rss_input_preprocess = function(sumstats, LD_data){
+    target_variants = sumstats[,c("chrom","pos","A1","A2")]
+    ref_variants = LD_data$combined_LD_variants
+    allele_flip = allele_qc(target_variants, ref_variants, sumstats, col_to_flip = c("beta", "z"), match.min.prop=0.2, remove_dups=FALSE, flip=TRUE, remove=TRUE)
+
+    sumstat_processed = allele_flip$target_data_qced %>% arrange(pos)
+    return(sumstat_processed)
+    
+}
+
+
+susie_rss_pipeline = function(sumstat, R, ref_panel, bhat = NULL, shat = NULL, n, L, var_y, QC = TRUE, impute = TRUE, cond_analysis = TRUE, rcond, R2_threshold, max_L, l_step, minimum_ld){
+        final_result = list()
+        z = sumstat$z
+        LD_extract = R[sumstat$variant_id, sumstat$variant_id, drop = FALSE]
+        single_effect_res = susie_rss_wrapper(z = z, R = LD_extract, bhat = NULL, shat = NULL, L = 1, n = n, var_y = var_y)
+        single_effect_post = susie_post_processor(single_effect_res, data_x = LD_extract, data_y = list(z), mode = "susie_rss")
+        final_result$single_effect_result = single_effect_post
+    
+        result_noqc = susie_rss_wrapper(z = z, R = LD_extract, bhat = NULL, shat = NULL, n = n, L = L, var_y = var_y)
+        result_noqc_post = susie_post_processor(result_noqc, data_x = LD_extract, data_y = list(z), mode = "susie_rss")
+        final_result$noqc_result = result_noqc_post
+
+
+        if(QC){
+            result_qced = susie_rss_qc(sumstat, ref_panel = ref_panel, R = R, n = n, L = L, impute = impute,
+            rcond = rcond, R2_threshold = R2_threshold, max_L = max_L, minimum_ld = minimum_ld, l_step = l_step, var_y = var_y)  
+            var_impute_kept = names(result_qced$qc_impute_result$pip)
+            
+            result_qced_impute_post = susie_post_processor(result_qced$qc_impute_result, data_x = R[var_impute_kept, var_impute_kept, drop = FALSE], data_y = list(z = result_qced$qc_impute_result$z), mode = "susie_rss")
+
+            result_qced_only_post = susie_post_processor(result_qced$qc_only_result, data_x = LD_extract, data_y = list(z = result_qced$qc_only_result$z), mode = "susie_rss")
+            
+            final_result$qc_impute_result = result_qced_impute_post
+            final_result$qc_only_result = result_qced_only_post
+            
+            result_qced$qc_impute_filter_table$chrom = as.numeric(result_qced$qc_impute_filter_table$chrom)
+            result_qced$qc_impute_nofilter_table$chrom = as.numeric(result_qced$qc_impute_nofilter_table$chrom)
+            final_result$qc_impute_filter_table = result_qced$qc_impute_filter_table
+            final_result$qc_impute_nofilter_table = result_qced$qc_impute_nofilter_table
+        }
+
+        if(cond_analysis){
```

Comment:
let's use the parameter name `bayesian_conditional_analysis` instead of `cond_analysis`